common/persistence/nosql/nosqlplugin/cassandra/tasks.go

```diff
@@ -89,10 +89,56 @@ func toTaskListPartitionConfig(v interface{}) *persistence.TaskListPartitionConf
 	version := partition["version"].(int64)
 	numRead := partition["num_read_partitions"].(int)
 	numWrite := partition["num_write_partitions"].(int)
+	readPartitions := toTaskListPartitions(partition["read_partitions"])
+	writePartitions := toTaskListPartitions(partition["write_partitions"])
+	// If they're out of sync, go with the value of num_*_partitions. This is necessary only while support for
+	// read_partitions and write_partitions rolls out
+	if numRead != len(readPartitions) {
+		readPartitions = createDefaultPartitions(numRead)
+	}
+	if numWrite != len(writePartitions) {
+		writePartitions = createDefaultPartitions(numWrite)
+	}
 	return &persistence.TaskListPartitionConfig{
-		Version:            version,
-		NumReadPartitions:  numRead,
-		NumWritePartitions: numWrite,
+		Version:         version,
+		ReadPartitions:  readPartitions,
+		WritePartitions: writePartitions,
+	}
+}
+
+func createDefaultPartitions(num int) map[int]*persistence.TaskListPartition {
+	partitions := make(map[int]*persistence.TaskListPartition, num)
+	for i := 0; i < num; i++ {
+		partitions[i] = &persistence.TaskListPartition{}
+	}
+	return partitions
+}
+
+func toTaskListPartitions(values any) map[int]*persistence.TaskListPartition {
+	if values == nil {
+		return nil
+	}
+	partitions := values.(map[int]map[string]any)
```

Comment:
nit: to avoid future misuses causing panics let's check the result of casting
```
partitions, ok := values.(map[int]map[string]any)
if !ok { return nil }
```

Reply:
Done.